vmm/src/config.rs

```diff
@@ -1223,6 +1246,10 @@ impl VmConfig {
             return Err(ValidationError::CpusMaxLowerThanBoot);
         }
 
+        if self.memory.file.is_some() {
+            return Err(ValidationError::MemoryFilePath);
+        }
```

Comment:
We need a check in the memory manager itself if we want to prevent OpenAPI from providing a `file`.

Reply:
The validation step is called on OpenAPI boots (hence why I added `validate`)

Reply:
No, actually we should do this
```
       if self.memory.file.is_some() {
            error!("Use of backing file is deprecated. Use the "shared" and "hugepages" controls");
        }
```

And then remove the file entry from the OpenAPI file and the configuration syntax after the next release. It should not be an error to supply it now.

Reply:
This makes sense, as the CLI and OpenAPI can still use it but it's deprecated.

Reply:
@rbradford @sboeuf  Right. We don't want to quit, and just need to log the error message. Now I understand better that the code path we added are parallel to the existing (deprecated) code path of using "file=". Just updated the  PR, and also the commit message. Please take another look. Thanks a lot.